packages/react-topology/src/components/TopologyView/examples/Topology.md

```diff
@@ -68,4 +69,14 @@ To do this, you will need:
 Within each node in your `NODES` array, you can set `data` to include additional custom attributes.
 
 ```ts file='./TopologyCustomNodeDemo.tsx'
+
+### Selectable Topology
+
+To make nodes and edges selectable:
+
+1. Create an intersection type using `WithSelectionProps` in the generic component parameters of `CustomNode`.
+1. Add `selected` and `onSelect` to the function parameters of `CustomNode`, and set them as values of the respectively named props in the `DefaultNode` component.
+1. In the `customComponentFactory` , set the `withSelection()` function on the return values of the nodes and edges.
```

Comment:
To allow nodes/edges to be selectable, you can use the `withSelection` utility when returning the component in the componentFactory: `withSelection(MyCustomNode)`

The component should except two parameters, `onSelect` and `selected` (you can simply extend `WithSelectionProps`). 
  - onSelect, function to call upon node selection. Typically the outer container for the component would call onSelect when clicked on
  - selected, indicates if the element is currently selected. Updated the drawing of the component to indicate its selection status.

Alternatively, you can use the `useSelection` hook within the component to retrieve the same two properties.

If you are using DefaultNode, these props can be passed along and will be handled appropriately.

By default, the application must control selection state. This can be done be adding a listener to the controller for a `SELECTION_EVENT`. The listener would then keep track of the selectedIds and maintain the ids in state and pass that state along to the `VisualizationSurface`. 

Alternatively, you can pass { controlled: true } to the `withSelection` utility or to the `useSelection` hook.